### ISSUE TITLE
Error	Firmware not found: 

### ISSUE THREAD

**Issue by chegevarko1982** [elral/MF_FCU_EFIS_OLEDs#5]
Error	01/22/2024 19:35:01	Firmware not found: E:\MobiFlight\MobiFlight Connector\Community\nomega\firmware\gagagu_fcu_efis_mega_0_9_0.hex.

in firmware folder we have reset.arduino_mega_1_0_2.hex only

**Comment by elral** [comment]
Did you download this zip file https://github.com/elral/MobiFlight-FCU_EFIS_OLEDs/releases/download/0.9.0/Gagagu_FCU_EFIS_0.9.0.zip ?
Just double checked that all files are included.
Where does the folder /nomega/ come from?

**Comment by chegevarko1982** [comment]
my bad i downloads from https://github.com/elral/MobiFlight-FCU_EFIS_OLEDs not from release bench

**Comment by chegevarko1982** [comment]
my bad i downloads from https://github.com/elral/MobiFlight-FCU_EFIS_OLEDs not from release bench

**Comment by elral** [comment]
@chegevarko1982 could you please give a short feedback once it's working for you? I guess you are using displays with SSD1306 driver I don't have. So I can be sure that these displays are also supported.

**Comment by chegevarko1982** [comment]
now firmware correct and flashing but again error
 Error	01/23/2024 18:51:34	avrdude timed out! Something went wrong when flashing with command 
 E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\bin\avrdude -C"E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\etc\avrdude.conf" -patmega2560 -cwiring -PCOM3 -b115200 -D -Uflash:w:"E:\MobiFlight\MobiFlight Connector\Community\Gagagu_FCU_EFIS\firmware\gagagu_fcu_efis_mega_0_9_0.hex":i.

**Comment by chegevarko1982** [comment]
![Без-имени-1](https://github.com/elral/MobiFlight-FCU_EFIS_OLEDs/assets/109864516/33dded0c-b3bb-4fb2-9611-264dff0020a8)
and

**Comment by elral** [comment]
Just open it. It checks the name of the board, and mine board had this one.
But I guess I will change it to mega or more general one.
Thanks for reporting.

**Comment by elral** [comment]
> now firmware correct and flashing but again error Error 01/23/2024 18:51:34 avrdude timed out! Something went wrong when flashing with command E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\bin\avrdude -C"E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\etc\avrdude.conf" -patmega2560 -cwiring -PCOM3 -b115200 -D -Uflash:w:"E:\MobiFlight\MobiFlight Connector\Community\Gagagu_FCU_EFIS\firmware\gagagu_fcu_efis_mega_0_9_0.hex":i.

And after rebooting?

**Comment by elral** [comment]
@chegevarko1982 please use this json file
https://github.com/elral/MobiFlight-FCU_EFIS_OLEDs/blob/increase_delay_time/FCU_EFIS/Community/boards/gagagu_fcu_efis_mega.board.json
Run into the same problem yesterday.

**Comment by elral** [comment]
Just merged it into the main branch.

**Comment by chegevarko1982** [comment]
![image](https://github.com/elral/MobiFlight-FCU_EFIS_OLEDs/assets/109864516/b66764ec-d72f-44db-8a82-2bca3b5186d8)
it was last distr

**Comment by elral** [comment]
Please enable debug mode under settings. Try it again and upload log.txt.

**Comment by chegevarko1982** [comment]
01/25/2024 19:07:52(743): avrdude timed out! Something went wrong when flashing with command 
 E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\bin\avrdude -C"E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\etc\avrdude.conf" -patmega2560 -cwiring -PCOM3 -b115200 -D -Uflash:w:"E:\MobiFlight\MobiFlight Connector\Community\erla_Gagagu_FCU_EFIS\firmware\gagagu_fcu_efis_mega_0_9_0.hex":i.
01/25/2024 19:14:44(537): avrdude timed out! Something went wrong when flashing with command 
 E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\bin\avrdude -C"E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\etc\avrdude.conf" -patmega2560 -cwiring -PCOM3 -b115200 -D -Uflash:w:"E:\MobiFlight\MobiFlight Connector\Community\erla_Gagagu_FCU_EFIS\firmware\gagagu_fcu_efis_mega_0_9_0.hex":i.

**Comment by elral** [comment]
It seems to be an issue from the connector that he does not wait long enough to be ready with the FW update.

I will close this issue for now.

**Comment by Alerido** [comment]
> 01/25/2024 19:07:52(743): avrdude timed out! Something went wrong when flashing with command E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\bin\avrdude -C"E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\etc\avrdude.conf" -patmega2560 -cwiring -PCOM3 -b115200 -D -Uflash:w:"E:\MobiFlight\MobiFlight Connector\Community\erla_Gagagu_FCU_EFIS\firmware\gagagu_fcu_efis_mega_0_9_0.hex":i. 01/25/2024 19:14:44(537): avrdude timed out! Something went wrong when flashing with command E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\bin\avrdude -C"E:\MobiFlight\MobiFlight Connector\Arduino\hardware\tools\avr\etc\avrdude.conf" -patmega2560 -cwiring -PCOM3 -b115200 -D -Uflash:w:"E:\MobiFlight\MobiFlight Connector\Community\erla_Gagagu_FCU_EFIS\firmware\gagagu_fcu_efis_mega_0_9_0.hex":i.

Same issue here :(

**Comment by elral** [comment]
@Alerido as I already mentioned, nothing which I could do on the firmware side. The upload of the FW takes so long as it's so big, that the connector times out.
Feel free to open an issue on the connector repo.

**Comment by Alerido** [comment]
![error](https://github.com/user-attachments/assets/e38631bd-fcb4-4e82-b1d0-02264311c530)


Hi, just FYI I tried to upload the firmware via ISP and I received this error, I'm not sure it's a connector problem.
Anyway I corrected Gagagu's code and with the esp32 everything works, tested for now on Fenix ​​and only the Displays.
As soon as I finish setting the inputs I will send everything to him.
Greetings
